kong/runloop/wasm.lua

```diff
@@ -231,24 +231,21 @@ end
 ---
 ---@return kong.runloop.wasm.filter_chain_reference? ref
 local function get_chain_ref(state, typ, service_id, route_id)
-  local ref
-
   if typ == TYPE_SERVICE and service_id then
-    ref = state.by_service[service_id]
+    return state.by_service[service_id]
+  end
 
-  elseif typ == TYPE_ROUTE and route_id then
-    ref = state.by_route[route_id]
+  if typ == TYPE_ROUTE and route_id then
+    return state.by_route[route_id]
+  end
 
-  elseif typ == TYPE_COMBINED and service_id and route_id then
+  if typ == TYPE_COMBINED and service_id and route_id then
     local routes = state.combined[service_id]
-    ref = routes and routes[route_id]
-
-  else
-    -- unreachable
-    error("unknown filter chain type: " .. tostring(typ), 2)
+    return routes and routes[route_id]
```

Comment:
I would probably prefer `if..elseif..else` but you can remove the local ref at the top and return ref at the bottom.